src/stores/Widgets/SpectralProfileWidget/SpectralProfileWidgetStore.ts

```diff
@@ -409,7 +400,17 @@ export class SpectralProfileWidgetStore extends RegionWidgetStore {
     }
 
     @computed get intensityOptions(): string[] {
-        return GetIntensityOptions(this.intensityConfig);
+        const frame = this.effectiveFrame;
+        const isMultiProfileActive = this.profileSelectionStore.activeProfileCategory === MultiProfileCategory.IMAGE;
+
+        const profiles = this.profileSelectionStore.profiles;
+        const displayedFrames = profiles.map(profile => profile.frame);
+
+        if (frame?.spectralReference && isMultiProfileActive) {
+            return frame.commonIntensityUnitWith(displayedFrames);
```

Comment:
I would prefer we add a function like `GetCommonIntensityOptions(configs: IntensityConfig[]): string[]` instead of using `commonIntensityUnitWith`. The function may become simpler, and we can add `frame.intensityConfig` to profiles instead of the entire `frame`.